### PR TITLE
ceph.spec.in: use devtoolset-9 for building crimson

### PR DESCRIPTION
**Pull request by tchaikov**
since seastar dropped support of C++14, we have to move to a compiler
with a decent C++17 support.

in this change, devtoolset-9 is used for compiling ceph if seastar is
enabled. use version >= 9.2.1-2.2, because of
https://bugzilla.redhat.com/show_bug.cgi?id=1853900

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
marking DNM, as it depends on gcc-toolset-9-gcc-9.2.1-2.3.el8 which is still not available at the time of writing.

File: ceph.spec.in [hunk 1]
```diff
@@ -1076,6 +1085,10 @@ This package provides Ceph’s default alerts for Prometheus.
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=48200
 %define _lto_cflags %{nil}
 
+%if 0%{?with seastar} == 8
```
Comment (ronen-fr) [inline ceph.spec.in:1088]: (just for my information...)  What is the '8' here? where is it set?

Comment (tchaikov) [reply ceph.spec.in:1088]: fixed and repushed.

**Comment by tchaikov** [comment]
see also https://github.com/ceph/ceph-build/pull/1641

**Comment by tchaikov** [comment]
changelog

- enable https://copr.fedorainfracloud.org/coprs/tchaikov/gcc-toolset-9 in install-deps.sh

**Pull request merged**